### PR TITLE
feat init SONOR app with mocked api

### PR DESCRIPTION
**Pull request by kc-bclaudel**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,61 @@
...
+Open [http://localhost:3000](http://localhost:3000) to view it in the browser.
+
+The page will reload if you make edits.<br />
+You will also see any lint errors in the console.
+
+### Start mock servers
+
+To test the application at this point you need to use the server *mocked_apis_server* to serve the pearlJam mocked APIs.
+
+Go to the *mocked_apis_server* folder and run
```
Comment (nicolasTurban) [inline README.md:32]: Can't find folder for running mocked api

File: README.md [hunk 2]
```diff
@@ -1,3 +1,61 @@
...
+
+`npm install` 
+
+`npm start` 
+
+You will also need to start the keycloak server configured for this app, in the *keycloak-sonor/bin* folder, run
```
Comment (nicolasTurban) [inline README.md:38]: Add keycloak server

File: README.md [hunk 3]
```diff
@@ -1,3 +1,61 @@
-# Sonor
+# Sonor app
 
-Case management web application for administration of Computer-Assisted Personal Interviewing (CAPI)
+## Installation
+### Requirements
+
+- Node.js
```
Comment (nicolasTurban) [inline README.md:6]: Versions ?

File: public/configuration.json [hunk 4]
```diff
@@ -0,0 +1,8 @@
+{
+  "PEARL_JAM_HOST": "http://localhost",
+  "PEARL_JAM_PORT": 7777,
+
+  "AUTHENTICATION_MODE": "keycloak",
+  "_AUTHENTICATION_MODE_COMMENT_": "Use 'keycloak' or 'no-auth'"
```
Comment (nicolasTurban) [inline public/configuration.json:6]: Prefer 'anonymous' and 'keycloak'

File: public/index.html [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+  <head>
+    <meta charset="utf-8" />
+    <link rel="icon" href="%PUBLIC_URL%/favicon.ico" />
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <meta name="theme-color" content="#000000" />
+    <meta
+      name="description"
+      content="Web site created using create-react-app"
+    />
+    <link rel="apple-touch-icon" href="%PUBLIC_URL%/logo192.png" />
```
Comment (nicolasTurban) [inline public/index.html:12]: Wrong logo

File: public/manifest.json [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+  "name": "Create React App Sample",
+  "icons": [
+    {
+      "src": "favicon.ico",
+      "sizes": "64x64 32x32 24x24 16x16",
+      "type": "image/x-icon"
+    },
+    {
+      "src": "logo192.png",
+      "type": "image/png",
```
Comment (nicolasTurban) [inline public/manifest.json:12]: Wrong logos

File: src/i18n/labels/labels_mainscreen.js [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+		"fr": "Confiées",
+		"en": "Allocated"
+	},
+	"toBeAssigned":{
+		"fr": "À affecter",
+		"en": "To be assigned"
+	},
+	"toTreatInterviewer":{
+		"fr": "À taiter enquêteur",
+		"en": "to treat by interviewer"
```
Comment (nicolasTurban) [inline src/i18n/labels/labels_mainscreen.js:40]: to -> To

File: src/utils/constants.json [hunk 8]
```diff
@@ -0,0 +1,11 @@
+{
+  "KEYCLOAK": "keycloak",
+  "NO_AUTH": "no-auth",
```
Comment (nicolasTurban) [inline src/utils/constants.json:3]: Prefer 'anonymous' as for queen app

File: README.md (hunk 1 above)
Comment (kc-bclaudel) [reply README.md:34]: Done. 
mocked_apis_server.zip added at root project

File: README.md (hunk 2 above)
Comment (kc-bclaudel) [reply README.md]: default authentication is now ANONYMOUS

File: README.md (hunk 3 above)
Comment (kc-bclaudel) [reply README.md]: DONE

File: public/configuration.json (hunk 4 above)
Comment (kc-bclaudel) [reply public/configuration.json]: Done

File: public/index.html (hunk 5 above)
Comment (kc-bclaudel) [reply public/index.html]: corrected

File: public/manifest.json (hunk 6 above)
Comment (kc-bclaudel) [reply public/manifest.json]: corrected

File: src/i18n/labels/labels_mainscreen.js (hunk 7 above)
Comment (kc-bclaudel) [reply src/i18n/labels/labels_mainscreen.js]: corrected

File: src/utils/constants.json (hunk 8 above)
Comment (kc-bclaudel) [reply src/utils/constants.json]: done

File: src/initConfiguration.js [hunk 9]
```diff
@@ -0,0 +1,13 @@
+const initConfiguration = async () => {
+  const publicUrl = new URL(process.env.PUBLIC_URL, window.location.href);
+  const response = await fetch(`${publicUrl.origin}/configuration.json`);
+  const configuration = await response.json();
+  window.localStorage.setItem(
+    'AUTHENTICATION_MODE',
+    configuration.AUTHENTICATION_MODE,
+  );
+  window.localStorage.setItem('PEARL_JAM_HOST', configuration.PEARL_JAM_HOST);
```
Comment (nicolasTurban) [inline src/initConfiguration.js:9]: No need for url splitting. Use PEARL_JAM_URL instead.

File: src/components/CampaignPortal/CampaignPortal.jsx [hunk 10]
```diff
@@ -0,0 +1,331 @@
...
+  );
+  return (
+    <Card className="ViewCard">
+      <div>
+        <Card.Title className="Title">{D.contacts}</Card.Title>
+
+        <Table className="CustomTable" bordered striped responsive size="sm">
+          <tbody>
+            <OverlayTrigger placement="top" overlay={renderTooltip}>
+              <tr className="Clickable" onClick={() => { window.location = 'mailto:gestion-enquete-mobilite@insee.fr'; }}>
```
Comment (nicolasTurban) [inline src/components/CampaignPortal/CampaignPortal.jsx:138]: prefer survey@mail.com

File: src/components/CampaignPortal/CampaignPortal.jsx [hunk 11]
```diff
@@ -0,0 +1,331 @@
...
+                <th>{D.functionalBox}</th>
+                <td className=" LightGreyLine">gestion-enquete-mobilite@insee.fr</td>
```
Comment (nicolasTurban) [inline src/components/CampaignPortal/CampaignPortal.jsx:140]: prefer survey@mail.com

File: src/components/CampaignPortal/CampaignPortal.jsx [hunk 12]
```diff
@@ -0,0 +1,331 @@
...
+              </tr>
+            </OverlayTrigger>
+            <tr>
+              <th rowSpan="2">{D.cpos}</th>
+              <td className="LightGreyLine">Chloé Berlin</td>
```
Comment (nicolasTurban) [inline src/components/CampaignPortal/CampaignPortal.jsx:145]: Chloé Dupont

File: src/components/CampaignPortal/CampaignPortal.jsx [hunk 13]
```diff
@@ -0,0 +1,331 @@
...
+            </tr>
+            <tr>
+              <td className="LightGreyLine">01 87 69 64 53</td>
```
Comment (nicolasTurban) [inline src/components/CampaignPortal/CampaignPortal.jsx:148]: 01 01 01 01 01

File: src/components/CampaignPortal/CampaignPortal.jsx [hunk 14]
```diff
@@ -0,0 +1,331 @@
...
+            </tr>
+            <tr>
+              <th rowSpan="2">{D.deputyCpos}</th>
+              <td className="LightGreyLine">Thierry Fabres</td>
+            </tr>
+            <tr>
+              <td className="LightGreyLine">01 22 55 13 09</td>
```
Comment (nicolasTurban) [inline src/components/CampaignPortal/CampaignPortal.jsx:155]: 02 01 01 01 01

File: mocked_apis_server/src/api/index.js [hunk 15]
```diff
@@ -0,0 +1,591 @@
...
+	let api = Router();
+
+	// mount the facets resource
+	api.use('/facets', facets({ config, db }));
+
+	//header
+	api.get('/user', (req, res) => {
+		const mockResponse = {
+		    firstName:"Chloé",
+		    lastName:"Berlin",
```
Comment (nicolasTurban) [inline mocked_apis_server/src/api/index.js:16]: Chloé Dupont

File: mocked_apis_server/src/api/index.js [hunk 16]
```diff
@@ -0,0 +1,591 @@
...
+	});
+
+
+	api.get('/campaign/:id/interviewers', (req, res) => {
+
+		const mockResponse = [
+		    {
+		        id:"INTW5",
+		        interviewerFirstName: "Chloé",
+		        interviewerLastName: "Berlin",
```
Comment (nicolasTurban) [inline mocked_apis_server/src/api/index.js:192]: Chloé Dupont

File: mocked_apis_server/src/api/index.js [hunk 17]
```diff
@@ -0,0 +1,591 @@
...
+
+		res.json(mockResponse);
+	});
+
+	api.get('/campaign/:id/contacts', (req, res) => {
+
+		const mockResponse = {
+		    cpos_first_name:"Chloé Berlin",
+		    cpos_last_name:"Chloé Berlin",
+		    cpos_phone_number: "01 87 69 64 53",
```
Comment (nicolasTurban) [inline mocked_apis_server/src/api/index.js:242]: Chloé Dupont 01 01 01 01 01

File: mocked_apis_server/src/api/index.js [hunk 18]
```diff
@@ -0,0 +1,591 @@
...
+		    },
+		    {
+		        id : "1024",
+		        ssech : "hgSkR29",
+		        location : "95160",
+		        city : "MONTMORENCY",
+		        interviewer : {
+		            id : "INTW5",
+		            firstName : "Chloé",
+		            lastName : "Berlin"
```
Comment (nicolasTurban) [inline mocked_apis_server/src/api/index.js:285]: Chloé Dupont

File: mocked_apis_server/src/api/index.js [hunk 19]
```diff
@@ -0,0 +1,591 @@
...
+		    },
+		    {
+		        id : "1025",
+		        ssech : "hgSkR29",
+		        location : "95160",
+		        city : "MONTMORENCY",
+		        interviewer : {
+		            id : "INTW5",
+		            firstName : "Chloé",
+		            lastName : "Berlin"
```
Comment (nicolasTurban) [inline mocked_apis_server/src/api/index.js:307]: Chloé Dupont

File: mocked_apis_server/src/api/index.js [hunk 20]
```diff
@@ -0,0 +1,591 @@
...
+		    },
+		    {
+		        id : "1027",
+		        ssech : "hgSkR29",
+		        location : "95160",
+		        city : "MONTMORENCY",
+		        interviewer : {
+		            id : "INTW5",
+		            firstName : "Chloé",
+		            lastName : "Berlin"
```
Comment (nicolasTurban) [inline mocked_apis_server/src/api/index.js:329]: Dupont

**Pull request merged**